aws/resource_aws_rds_cluster_endpoint.go

```diff
@@ -0,0 +1,232 @@
+	}
+	log.Printf("[DEBUG] Describing RDS Cluster: %s", input)
+	resp, err := conn.DescribeDBClusterEndpoints(input)
+
+	if err != nil {
+		return fmt.Errorf("error describing RDS Cluster Endpoints (%s): %s", d.Id(), err)
+	}
+
+	if resp == nil {
+		return fmt.Errorf("Error retrieving RDS Cluster Endpoints: empty response for: %s", input)
+	}
+
+	var clusterEp *rds.DBClusterEndpoint
+	for _, e := range resp.DBClusterEndpoints {
+		if aws.StringValue(e.DBClusterEndpointIdentifier) == d.Id() {
+			clusterEp = e
+			break
+		}
+	}
+
+	if clusterEp == nil {
+		log.Printf("[WARN] RDS Cluster Endpoint (%s) not found", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("cluster_endpoint_identifier", clusterEp.DBClusterEndpointIdentifier)
+	d.Set("cluster_identifier", clusterEp.DBClusterIdentifier)
+	d.Set("arn", clusterEp.DBClusterEndpointArn)
+	d.Set("endpoint", clusterEp.Endpoint)
+	d.Set("custom_endpoint_type", clusterEp.CustomEndpointType)
+
+	excludeMembers := make([]string, 0)
+	for _, member := range clusterEp.ExcludedMembers {
+		excludeMembers = append(excludeMembers, aws.StringValue(member))
+	}
+	d.Set("excluded_members", excludeMembers)
+
+	staticMembers := make([]string, 0)
+	for _, member := range clusterEp.StaticMembers {
+		staticMembers = append(staticMembers, aws.StringValue(member))
+	}
+	d.Set("static_members", staticMembers)
+
+	return nil
+}
+
+func resourceAwsRDSClusterEndpointUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	input := &rds.ModifyDBClusterEndpointInput{
+		DBClusterEndpointIdentifier: aws.String(d.Id()),
+	}
+
+	if v, ok := d.GetOk("custom_endpoint_type"); ok {
+		input.EndpointType = aws.String(v.(string))
+	}
+
+	if attr := d.Get("excluded_members").(*schema.Set); attr.Len() > 0 {
+		input.ExcludedMembers = expandStringList(attr.List())
+	} else {
+		input.ExcludedMembers = make([]*string, 0)
+	}
+
+	if attr := d.Get("static_members").(*schema.Set); attr.Len() > 0 {
+		input.StaticMembers = expandStringList(attr.List())
+	} else {
+		input.StaticMembers = make([]*string, 0)
+	}
+
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := conn.ModifyDBClusterEndpoint(input)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Error modifying RDS Cluster Endpoint: %s", err)
+	}
+
+	return resourceAwsRDSClusterEndpointRead(d, meta)
+}
+
+func resourceAwsRDSClusterEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	input := &rds.DeleteDBClusterEndpointInput{
+		DBClusterEndpointIdentifier: aws.String(d.Id()),
+	}
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := conn.DeleteDBClusterEndpoint(input)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Error deleting RDS Cluster Endpoint: %s", err)
+	}
+	return nil
```

Comment:
Similar to the `Create` and `Update` functions -- the `DBClusterEndpoint` struct in the AWS SDK has a `Status` field with this documentation:

```
	// The current status of the endpoint. One of: creating, available, deleting,
	// modifying.
	Status *string `type:"string"`
```

It seems we should be waiting until its fully deleted. See also other examples of `StateChangeConf` in the codebase for examples of how to accomplish this.